src/main/java/com/example/ticketbooking/service/RowService.java

```diff
@@ -0,0 +1,38 @@
+package com.example.ticketbooking.service;
+
+import com.example.ticketbooking.exception.EntityNotFoundException;
+import com.example.ticketbooking.model.Row;
+import com.example.ticketbooking.model.request.ReservationRequest;
+import lombok.experimental.UtilityClass;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@UtilityClass
+public class RowService {
```

Comment:
No to w końcu Service czy Utils? :)

Reply:
Szczerze to nie jestem pewien jeśli chodzi o nazewnictwo. `Utility` kojarzy mi się z klasą, która daje jakieś dodatkowe funkcje niezwiązane bezpośrednio z naszymi klasami domenowymi - TimeUtils, EncodingUtils, czy coś. Natomiast jeśli jest klasa, która operuje na naszym modelu, ale ma akurat wszystkie metody statyczne, to bardziej mi tu `service` pasuje, choć może moje myślenie jest błędne :P

Reply:
Pierwsza metoda, czyli `createRows` bardziej by pasowała do wzorca fabryki. Natomiast metoda sprawdzająca zajętość miejsc mogłaby być spokojnie w klasie `Row`, ale nie tej encji z DB, tylko w warstwie domenowej.